cmd/nerdctl/volume_prune.go

```diff
@@ -0,0 +1,115 @@
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/containerd/nerdctl/pkg/containerinspector"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+func newVolumePruneCommand() *cobra.Command {
+	volumeRmCommand := &cobra.Command{
+		Use:           "prune",
+		Short:         "Remove one or more volumes",
+		Long:          "NOTE: volume in use is not deleted",
+		RunE:          volumePruneAction,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+	volumeRmCommand.Flags().BoolP("force", "f", false, "Do not prompt for confirmation")
+	return volumeRmCommand
+}
+
+func volumePruneAction(cmd *cobra.Command, args []string) error {
+	client, ctx, cancel, err := newClient(cmd)
+	if err != nil {
+		return err
+	}
+	defer cancel()
+	force, err := cmd.Flags().GetBool("force")
+	if err != nil {
+		return err
+	}
+	if !force {
+		var confirm string
+		fmt.Fprintf(cmd.OutOrStdout(), "%s", "WARNING! This will remove all local volumes not used by at least one container.\nAre you sure you want to continue? [y/N] ")
+		fmt.Fscanf(cmd.InOrStdin(), "%s", &confirm)
+
+		if strings.ToLower(confirm) != "y" {
+			return nil
+		}
+	}
+	containers, err := client.Containers(ctx)
+	if err != nil {
+		return err
+	}
+	var mount *specs.Spec
+	var removednames []string
+	volumes, err := getVolumes(cmd)
+	if err != nil {
+		return err
+	}
+	volStore, err := getVolumeStore(cmd)
+	if err != nil {
+		return err
+	}
+	for name, voldetails := range volumes {
+		var found bool
+		for _, container := range containers {
+			n, err := containerinspector.Inspect(ctx, container)
+			if err != nil {
+				return err
+			}
+			err = json.Unmarshal(n.Container.Spec.Value, &mount)
+			if err != nil {
+				return err
+			}
+			if found = checkVolume(&mount.Mounts, voldetails.Mountpoint); found {
+				found = true
+				logrus.WithError(fmt.Errorf("Volume %q is in use", name)).Error(fmt.Errorf("Remove Volume: %q failed", name))
+				break
+			}
+		}
+		if !found {
+			// volumes that are not mounted to any container
+			removednames = append(removednames, name)
+		}
+	}
+	if removednames != nil {
+		rNames, err := volStore.Remove(removednames)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintln(cmd.OutOrStdout(), "Deleted Volumes")
+		print(rNames, cmd)
+	}
+	return nil
+}
+
+func print(x []string, cmd *cobra.Command) {
```

Comment:
This conflicts with built-in `print`.
Please come up with a unique and descriptive name.

Reply:
Okay

Reply:
> Display

"Display" isn't unique/descriptive, and I still don't get what this function does and why this function is needed.

Reply:
This function displays the containers one-by-one in this way:
```
Deleted Volumes
testVolume123
testVolume
testVolume007
```
and not in this way:
```
Deleted Volumes: [testVolume123 testVolume testVolume007]
```